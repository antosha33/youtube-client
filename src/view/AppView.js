import './main.css';

export default class Appview {
  constructor(startQuestion) {
    this.startQuestion = startQuestion;
    this.result = [];
  }

  startRender() {
    const wrapper = document.createElement('div');
    const search = document.createElement('div');
    const label = document.createElement('label');
    const content = document.createElement('input');
    const loadingLogo = document.createElement('div');
    loadingLogo.classList.add('loading-logo');
    wrapper.classList.add('wrapper');
    wrapper.setAttribute('id', 'wrapper');
    search.classList.add('search-wrapper');
    label.appendChild(content);
    content.setAttribute('placeholder', this.startQuestion);
    content.setAttribute('name', 'main-search');
    content.setAttribute('id', 'main-input');
    search.appendChild(label);
    wrapper.appendChild(search);
    wrapper.appendChild(loadingLogo);
    document.body.appendChild(wrapper);
  }

  resultRender(data) {
    this.clearFlow();
    if (document.getElementById('clips')) document.getElementById('clips').remove();
    this.result = data;
    const ul = document.createElement('ul');
    ul.classList.add('clip-list');
    ul.setAttribute('id', 'clips');
    ul.innerHTML = this.result.map(item => `<li>
      <div class='item-wrapper'>
        <img class='clip-logo' src='${item.image}'>
        <a href='https://www.youtube.com/watch?v=${item.id}' target='_blank'><div class='clip-title'>${item.title}</div></a>
        <div class='clip-about'>
          <div class='clip-author'>${item.author}</div>
          <div class='clip-date'>${item.uploadDate.slice(0, -14)}</div>
          <div class='clip-views'>${item.view}</div>
          <div class='clip-description'>${item.description}</div>
        </div>
      </div>
    </li>
    `).join('');
    document.getElementById('wrapper').appendChild(ul);
    const navbar = document.createElement('div');
    navbar.classList.add('navbar');
    navbar.setAttribute('id', 'navbar');
    navbar.innerHTML = '<div class="prevPrev" id="prevPrev"></div><div class="prev" id="prev"></div><div class="page" id="page"><span>1</span></div><div class="next" id="next"</div>';
    document.getElementById('wrapper').appendChild(navbar);
  }

  addResults(data) {
    this.result = data;
    const clips = document.getElementById('clips');
    const liArray = this.result.map(item => `
      <div class='item-wrapper'>
      <img class='clip-logo' src='${item.image}'>
      <a href='https://www.youtube.com/watch?v=${item.id}' target='_blank'><div class='clip-title'>${item.title}</div></a>
        <div class='clip-about'>
        <div class='clip-author'>${item.author}</div>
        <div class='clip-date'>${item.uploadDate.slice(0, -14)}</div>
        <div class='clip-views'>${item.view}</div>
        <div class='clip-description'>${item.description}</div>
        </div>
      </div>`);
    liArray.forEach((it) => {
      const li = document.createElement('li');
      li.innerHTML = it;
      clips.appendChild(li);
    });
  }

  clearFlow() {
    this.result = [];
    if (document.getElementById('clips')) document.getElementById('clips').remove();
    if (document.getElementById('navbar')) document.getElementById('navbar').remove();
  }
}
