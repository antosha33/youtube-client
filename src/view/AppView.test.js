import AppView from './AppView';

describe('AppView.startRender && AppView.resultRender', () => {
  it('Should be rendered correctly', () => {
    const context1 = {
      startQuestion: 'What do you want to find ?',
    };
    const context = [
      {
        author: 'Author_1',
        description: 'description_1',
        id: 'id_1',
        image: 'image_1',
        title: 'title_1',
        uploadDate: 'uploadDate_1',
        view: 'view_1',
      },
      {
        author: 'Author_2',
        description: 'description_2',
        id: 'id_2',
        image: 'image_2',
        title: 'title_2',
        uploadDate: 'uploadDate_2',
        view: 'view_2',
      },
      {
        author: 'Author_3',
        description: 'description_3',
        id: 'id_3',
        image: 'image_3',
        title: 'title_3',
        uploadDate: 'uploadDate_3',
        view: 'view_3',
      },
      {
        author: 'Author_4',
        description: 'description_4',
        id: 'id_4',
        image: 'image_4',
        title: 'title_4',
        uploadDate: 'uploadDate_4',
        view: 'view_4',
      },
    ];
    AppView.prototype.startRender.call(context1);
    AppView.prototype.startRender(context);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
