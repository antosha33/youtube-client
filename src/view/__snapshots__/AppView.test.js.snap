// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.startRender && AppView.resultRender Should be rendered correctly 1`] = `"<div class=\\"wrapper\\" id=\\"wrapper\\"><div class=\\"search-wrapper\\"><label><input placeholder=\\"What do you want to find ?\\" name=\\"main-search\\" id=\\"main-input\\"></label></div><div class=\\"loading-logo\\"></div></div><div class=\\"wrapper\\" id=\\"wrapper\\"><div class=\\"search-wrapper\\"><label><input placeholder=\\"undefined\\" name=\\"main-search\\" id=\\"main-input\\"></label></div><div class=\\"loading-logo\\"></div></div>"`;
